src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetDelegateForFunctionPointerTests.cs

```diff
@@ -75,6 +76,17 @@ public void GetDelegateForFunctionPointer_GenericInvalidType_ReturnsExpected()
             VerifyDelegate(functionDelegate, targetMethod);
         }
 
+        [Fact]
+        public void GetDelegateForFunctionPointer_DelegateBaseTypes_ThrowsMustBeDelegate()
```

Comment:
Added test coverage for the path that I am touching. I expect that this test will fail on Mono (will file bug).